app/assets/javascripts/discourse/widgets/search-menu-results.js.es6

```diff
@@ -46,7 +47,8 @@ function postResult(result, link, term) {
 }
 
 createSearchResult('user', 'path', function(u) {
-  return [ avatarImg('small', { template: u.avatar_template, username: u.username }), ' ', u.username ];
+  console.log(u);
+  return [ avatarImg('small', { template: u.avatar_template, username: u.username }), ' ', h('span.user-results', h('b', u.username)), ' ',  h('span.user-results', u.name ? u.name : '') ];
```

Comment:
Here i'm checking if `name` is present in the attributes i got from the serializer. If is present it will show, if is not, it will not.
If `enable_name` is disable, server or client side, it will not be passed as attribute which will result as `false` for `u.name`.  
I have tested with `enable_name` enable/disable from client and it works correctly.
I also tested if user provide or not the name, at it also works correctly.
Do you maybe mean another case?